File: 406.根据身高重建队列.cpp
/*
 * @lc app=leetcode.cn id=406 lang=cpp
 *
 * [406] 根据身高重建队列
 */

// @lc code=start
class Solution {
public:
    vector<vector<int>> reconstructQueue(vector<vector<int>>& people) {
        sort(people.begin(), people.end(), [ ]( const auto& lhs, const auto& rhs ) {
            if (lhs[0] > rhs[0]) {
                return true;
            } else if (lhs[0] < rhs[0]) {
                return false;
            } else {
                return lhs[1] < rhs[1];
            }
        });
        vector<vector<int>> result;
        for (auto &person : people) {
            result.insert(result.begin() + person[1], person);
        }
        return result;
    }
};
// @lc code=end

File: 406.根据身高重建队列.swift
/*
 * @lc app=leetcode.cn id=406 lang=swift
 *
 * [406] 根据身高重建队列
 */

// @lc code=start
class Solution {
    func reconstructQueue(_ people: [[Int]]) -> [[Int]] {
        var peoples = people.sorted(by: { (lhs, rhs) -> Bool in
            if lhs[0] == rhs[0] {
                return lhs[1] < rhs[1]
            } else {
                return lhs[0] > rhs[0]
            }
        })
        var res = [[Int]]()
        for person in peoples {
            res.insert(person, at: person[1])
        }
        return res
    }
}
// @lc code=end
